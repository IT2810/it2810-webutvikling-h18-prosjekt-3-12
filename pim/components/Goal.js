import React from "react";
import { StyleSheet, Text, TouchableOpacity, View, Alert } from "react-native";
import { Icon } from "expo";
import Modal from "react-native-modal";
import { AsyncStorage } from "react-native";

class Goal extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      asyncKey: this.props.asyncKey,
      name: this.props.name,
      desc: this.props.desc,
      date: this.props.date,
      isModalVisible: false,
      checked: false
    };
  }

  //DEPRECATED, REPLACE
  //componentWillReceiveProps(nextProps) {
  // update original states
  //this.setState({
  //  index: nextProps.index,
  //  name: nextProps.name,
  //  text: nextProps.text,
  //  date: nextProps.date,
  //  isModalVisible: false
  //});
  //}

  //limit the length of descritpion to 20 character. For more the user can press info to view full desc
  limitDesc = () => {
    if (this.state.desc.length > 20) {
      return this.state.desc.substring(0, 20) + "...";
    }
    return this.state.desc;
  };

  //toogles the modal for the goal on/off
  toggleModal = () =>
    this.setState({ isModalVisible: !this.state.isModalVisible });

  //swithes checked state and performs corresponding action to asyncstorage
  handleCheckBoxToogle = () => {
    if (!this.state.checked) {
      this.setState({ checked: !this.state.checked });
      this.removeItemValue(this.state.asyncKey);
      this.toggleModal();
      this.refCallback();
    }
  };

  //removes item from async based on key
  async removeItemValue(key) {
    try {
      await AsyncStorage.removeItem(key);
    } catch (exception) {
      console.log("Error removing item", exception);
    }
  }

  //callback to Goal.js with update for removed item
  refCallback = element => {
    this.props.numGoals(-1);
  };

  render() {
    return (
      <View>
        <TouchableOpacity onPress={this.toggleModal}>
          <View
            style={
              this.state.checked === true
                ? styles.singleGoalContainerChecked
                : styles.singleGoalContainer
            }
          >
            <View style={styles.goalsNameContainer}>
              <Text
                style={
                  this.state.checked === true
                    ? styles.goalHeaderTextChecked
                    : styles.goalHeaderText
                }
              >
                {this.state.name}
              </Text>
              <Icon.Ionicons
                name="ios-information-circle-outline"
                size={16}
                style={{ color: "rgba(96,100,109, 1)" }}
              />
            </View>
            <View style={styles.goalsInfoContainer}>
              <Text style={styles.goalDescText}>{this.limitDesc()} </Text>
              <Text style={styles.goalDateText}>{this.state.date}</Text>
            </View>
          </View>
        </TouchableOpacity>
        <Modal
          isVisible={this.state.isModalVisible}
          backdropOpacity={0.7}
          onBackdropPress={this.toggleModal}
        >
          <View style={styles.modal}>
            <View style={styles.headerContainer}>
              <Text style={styles.modalTextHeader}>{this.state.name}</Text>
              <TouchableOpacity onPress={this.toggleModal}>
                <Icon.Ionicons
                  name="ios-close-circle-outline"
                  size={20}
                  style={{ color: "black", padding: 10 }}
                />
              </TouchableOpacity>
            </View>
            <Text style={styles.modalTextDesc}>{this.state.desc}</Text>
            <View style={styles.modalBottomColumn}>
              <View style={styles.modalBottomRow}>
                <TouchableOpacity onPress={this.handleCheckBoxToogle}>
                  <View style={styles.completedContainer}>
                    <Icon.Ionicons
                      name={
                        this.state.checked === false
                          ? "ios-radio-button-off"
                          : "ios-radio-button-on"
                      }
                      size={16}
                      style={{ color: "rgba(96,100,109, 1)", marginTop: 3 }}
                    />
                    <Text style={styles.modalTextComplete}>Completed</Text>
                  </View>
                </TouchableOpacity>
                <Text style={styles.modalTextDate}>{this.state.date}</Text>
              </View>
            </View>
          </View>
        </Modal>
      </View>
    );
  }
}

export default Goal;

const styles = StyleSheet.create({
  goalsNameContainer: {
    flex: 1,
    flexDirection: "row",
    justifyContent: "space-between"
  },
  goalsInfoContainer: {
    flex: 1,
    flexDirection: "row",
    justifyContent: "space-between"
  },
  goalDescText: {
    color: "rgba(96,100,109, 1)",
    lineHeight: 24
  },
  goalDateText: {
    fontSize: 13,
    color: "rgba(96,100,109, 1)"
  },
  singleGoalContainer: {
    borderRadius: 7,
    borderWidth: 0.5,
    borderColor: "#d6d7da",
    padding: 10,
    marginBottom: 10,
    backgroundColor: "#E9F7FD",
    overflow: "hidden"
  },
  singleGoalContainerChecked: {
    borderRadius: 7,
    borderWidth: 0.5,
    borderColor: "#d6d7da",
    padding: 10,
    marginBottom: 10,
    backgroundColor: "#E9F7FD",
    opacity: 0.4,
    overflow: "hidden"
  },
  goalHeaderTextChecked: {
    fontSize: 16,
    color: "black",
    lineHeight: 24,
    textDecorationLine: "line-through"
  },
  goalHeaderText: {
    fontSize: 16,
    color: "black",
    lineHeight: 24,
    textDecorationLine: "none"
  },
  modal: {
    marginHorizontal: 20,
    minHeight: 200,
    maxHeight: 300,
    backgroundColor: "white",
    borderRadius: 7,
    borderWidth: 0.5,
    borderColor: "#d6d7da"
  },
  headerContainer: {
    flex: 1,
    flexDirection: "row",
    justifyContent: "space-between"
  },
  modalBottomColumn: {
    padding: 10,
    flex: 1,
    flexDirection: "column",
    justifyContent: "flex-end"
  },
  modalBottomRow: {
    flexDirection: "row",
    justifyContent: "space-between"
  },
  modalTextHeader: {
    paddingLeft: 10,
    paddingRight: 5,
    paddingTop: 15,
    fontSize: 17,
    color: "black",
    lineHeight: 24
  },
  modalTextDesc: {
    paddingLeft: 10,
    fontSize: 16,
    color: "rgba(96,100,109, 1)",
    lineHeight: 24,
    marginTop: -30
  },
  modalTextComplete: {
    fontSize: 16,
    paddingRight: 5,
    color: "rgba(96,100,109, 1)",
    lineHeight: 24,
    paddingLeft: 7
  },
  modalTextDate: {
    fontSize: 12,
    color: "rgba(96,100,109, 1)",
    marginTop: 5
  },
  completedContainer: {
    flexDirection: "row"
  }
});
