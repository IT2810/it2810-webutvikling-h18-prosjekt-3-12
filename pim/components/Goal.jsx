import React, { Component } from 'react';
import {
    Image,
    Platform,
    ScrollView,
    StyleSheet,
    Text,
    TouchableOpacity,
    View,
    Alert
  } from "react-native";


export class Goal extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            text : "",
            date : Date(),
            index : props.index,
            name : ""
          }
    }
    render() { 
        return ( 
        <View>
            <ScrollView style={styles.container}>
                <View> // className="Index"
                    <Text>
                        {this.state.index}
                    </Text>
                </View>
                <View> // className="Name"
                    <Text>
                        {this.state.name}
                    </Text>
                </View>
                <View> // className="Date"
                    <Text>
                        {this.state.date}
                    </Text>
                </View>
            </ScrollView>
        </View>
         );
    }


}
 