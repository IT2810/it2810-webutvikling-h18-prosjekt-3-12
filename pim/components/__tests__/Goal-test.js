import "react-native";
import React from "react";
import Goal from "../Goal.js";
import renderer from "react-test-renderer";
import ShallowRenderer from "react-test-renderer/shallow";

describe("Goal snapshot", () => {
  jest.useFakeTimers();
  beforeEach(() => {});

  const goal = (
    <Goal
      asyncKey={"10394234234"}
      name={"Løp 1000m"}
      desc={"Mot opp ved lerkendal stadion og løp med Anders"}
      date={"Oct 3, 11:32"}
      numGoals={5}
    />
  );

  it("It should render the whole Goal component correctly", () => {
    const tree = renderer.create(goal).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should shallowrender Goal component correctly", () => {
    const renderer = new ShallowRenderer();
    const result = renderer.render(goal);
    expect(result).toMatchSnapshot();
  });
});
