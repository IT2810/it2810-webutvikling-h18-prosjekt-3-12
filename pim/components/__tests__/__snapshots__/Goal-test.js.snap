// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Goal snapshot It should render the whole Goal component correctly 1`] = `
<View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#E9F7FD",
          "borderColor": "#d6d7da",
          "borderRadius": 7,
          "borderWidth": 0.5,
          "marginBottom": 10,
          "overflow": "hidden",
          "padding": 10,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "black",
              "fontSize": 16,
              "lineHeight": 24,
              "textDecorationLine": "none",
            }
          }
        >
          Løp 1000m
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        />
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "rgba(96,100,109, 1)",
              "lineHeight": 24,
            }
          }
        >
          Mot opp ved lerkenda...
           
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "rgba(96,100,109, 1)",
              "fontSize": 13,
            }
          }
        >
          Oct 3, 11:32
        </Text>
      </View>
    </View>
  </View>
  <Modal
    animationType="none"
    hardwareAccelerated={false}
    hideModalContentWhileAnimating={false}
    onModalHide={[Function]}
    onRequestClose={[Function]}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
    transparent={true}
    visible={false}
  >
    <View
      accessible={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "black",
          "bottom": 0,
          "height": 1334,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 750,
        }
      }
    />
    <View
      hideModalContentWhileAnimating={false}
      onModalHide={[Function]}
      pointerEvents="box-none"
      scrollOffset={0}
      scrollOffsetMax={0}
      scrollTo={null}
      style={
        Object {
          "flex": 1,
          "justifyContent": "center",
          "margin": 37.5,
          "transform": Array [
            Object {
              "translateY": 0,
            },
          ],
        }
      }
      supportedOrientations={
        Array [
          "portrait",
          "landscape",
        ]
      }
      swipeThreshold={100}
    >
      <View
        style={
          Object {
            "backgroundColor": "white",
            "borderColor": "#d6d7da",
            "borderRadius": 7,
            "borderWidth": 0.5,
            "marginHorizontal": 20,
            "maxHeight": 300,
            "minHeight": 200,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "black",
                "fontSize": 17,
                "lineHeight": 24,
                "paddingLeft": 10,
                "paddingRight": 5,
                "paddingTop": 15,
              }
            }
          >
            Løp 1000m
          </Text>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            />
          </View>
        </View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "rgba(96,100,109, 1)",
              "fontSize": 16,
              "lineHeight": 24,
              "marginTop": -30,
              "paddingLeft": 10,
            }
          }
        >
          Mot opp ved lerkendal stadion og løp med Anders
        </Text>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "column",
              "justifyContent": "flex-end",
              "padding": 10,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "flexDirection": "row",
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                />
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "rgba(96,100,109, 1)",
                      "fontSize": 16,
                      "lineHeight": 24,
                      "paddingLeft": 7,
                      "paddingRight": 5,
                    }
                  }
                >
                  Completed
                </Text>
              </View>
            </View>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "rgba(96,100,109, 1)",
                  "fontSize": 12,
                  "marginTop": 5,
                }
              }
            >
              Oct 3, 11:32
            </Text>
          </View>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;

exports[`Goal snapshot should shallowrender Goal component correctly 1`] = `
<Component>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
  >
    <Component
      style={
        Object {
          "backgroundColor": "#E9F7FD",
          "borderColor": "#d6d7da",
          "borderRadius": 7,
          "borderWidth": 0.5,
          "marginBottom": 10,
          "overflow": "hidden",
          "padding": 10,
        }
      }
    >
      <Component
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "black",
              "fontSize": 16,
              "lineHeight": 24,
              "textDecorationLine": "none",
            }
          }
        >
          Løp 1000m
        </Text>
        <Icon
          allowFontScaling={false}
          name="ios-information-circle-outline"
          size={16}
          style={
            Object {
              "color": "rgba(96,100,109, 1)",
            }
          }
        />
      </Component>
      <Component
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "rgba(96,100,109, 1)",
              "lineHeight": 24,
            }
          }
        >
          Mot opp ved lerkenda...
           
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "rgba(96,100,109, 1)",
              "fontSize": 13,
            }
          }
        >
          Oct 3, 11:32
        </Text>
      </Component>
    </Component>
  </TouchableOpacity>
  <ReactNativeModal
    animationIn="slideInUp"
    animationInTiming={300}
    animationOut="slideOutDown"
    animationOutTiming={300}
    avoidKeyboard={false}
    backdropColor="black"
    backdropOpacity={0.7}
    backdropTransitionInTiming={300}
    backdropTransitionOutTiming={300}
    hideModalContentWhileAnimating={false}
    isVisible={false}
    onBackButtonPress={[Function]}
    onBackdropPress={[Function]}
    onModalHide={[Function]}
    onModalShow={[Function]}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
    useNativeDriver={false}
  >
    <Component
      style={
        Object {
          "backgroundColor": "white",
          "borderColor": "#d6d7da",
          "borderRadius": 7,
          "borderWidth": 0.5,
          "marginHorizontal": 20,
          "maxHeight": 300,
          "minHeight": 200,
        }
      }
    >
      <Component
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "black",
              "fontSize": 17,
              "lineHeight": 24,
              "paddingLeft": 10,
              "paddingRight": 5,
              "paddingTop": 15,
            }
          }
        >
          Løp 1000m
        </Text>
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
        >
          <Icon
            allowFontScaling={false}
            name="ios-close-circle-outline"
            size={20}
            style={
              Object {
                "color": "black",
                "padding": 10,
              }
            }
          />
        </TouchableOpacity>
      </Component>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "rgba(96,100,109, 1)",
            "fontSize": 16,
            "lineHeight": 24,
            "marginTop": -30,
            "paddingLeft": 10,
          }
        }
      >
        Mot opp ved lerkendal stadion og løp med Anders
      </Text>
      <Component
        style={
          Object {
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "flex-end",
            "padding": 10,
          }
        }
      >
        <Component
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
          >
            <Component
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            >
              <Icon
                allowFontScaling={false}
                name="ios-radio-button-off"
                size={16}
                style={
                  Object {
                    "color": "rgba(96,100,109, 1)",
                    "marginTop": 3,
                  }
                }
              />
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "rgba(96,100,109, 1)",
                    "fontSize": 16,
                    "lineHeight": 24,
                    "paddingLeft": 7,
                    "paddingRight": 5,
                  }
                }
              >
                Completed
              </Text>
            </Component>
          </TouchableOpacity>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "rgba(96,100,109, 1)",
                "fontSize": 12,
                "marginTop": 5,
              }
            }
          >
            Oct 3, 11:32
          </Text>
        </Component>
      </Component>
    </Component>
  </ReactNativeModal>
</Component>
`;
